File: library/src/main/java/io/vithor/yamvpframework/migration/SampleVersionManager.java
package io.vithor.yamvpframework.migration;

/**
 * Created by Vithorio Polten on 2/23/16.
 */
public final class SampleVersionManager extends AbstractVersionManager {
    /**
     * DO NOT SUBCLASS THIS, SAMPLE ONLY CLASS
     */
    private static SampleVersionManager instance = new SampleVersionManager();

    public static SampleVersionManager getInstance() {
        return instance;
    }

    private SampleVersionManager() {
        super();
        addMigrations(
                new V1_TestMigration()
        );
    }
}

File: migration/src/main/java/io/vithor/yamvpframework/migration/SampleVersionManager.kt
package io.vithor.yamvpframework.migration

/**
 * Created by Vithorio Polten on 2/23/16.
 */
class SampleVersionManager private constructor() : AbstractVersionManager() {

    init {
        addMigrations(
                V1_TestMigration())
    }

    companion object {
        /**
         * DO NOT SUBCLASS THIS, SAMPLE ONLY CLASS
         */
        val instance = SampleVersionManager()
    }
}